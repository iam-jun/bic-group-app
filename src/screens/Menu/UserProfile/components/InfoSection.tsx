import React from 'react';
import { StyleSheet, View } from 'react-native';
import Text from '~/beinComponents/Text';
import { spacing } from '~/theme';

interface Props {
  title: string;
  children: React.ReactNode
}

const InfoSection = ({ title, children }: Props) => (
  <View style={styles.container}>
    <Text.H4 useI18n style={styles.title}>{title}</Text.H4>
    {children}
  </View>
)
const styles = StyleSheet.create({
  container: {
    paddingBottom: spacing.padding.base,
    marginTop: spacing.margin.large,
  },
  title: {
    marginBottom: spacing.margin.large,
  },
});

export default InfoSection;
