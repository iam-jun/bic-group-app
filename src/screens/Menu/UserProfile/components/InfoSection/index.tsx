import React, { ReactElement } from 'react';
import { StyleSheet, View } from 'react-native';
import Text from '~/baseComponents/Text';
import { spacing } from '~/theme';

interface Props {
  testID?: string;
  title: string;
  rightTitle?: ReactElement;
  children: React.ReactNode;
}

const InfoSection = ({
  testID, title, rightTitle, children,
}: Props) => (
  <View testID={testID || 'info_section'}>
    <View style={styles.row}>
      <Text.H4 testID="info_section.title" useI18n>
        {title}
      </Text.H4>
      {rightTitle}
    </View>
    {children}
  </View>
);

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginBottom: spacing.margin.large,
  },
});

export default InfoSection;
