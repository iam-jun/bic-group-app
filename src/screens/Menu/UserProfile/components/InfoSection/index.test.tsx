import * as React from 'react';
import { render, cleanup } from '@testing-library/react-native';

import InfoSection from '.';

describe('InfoSection component', () => {
  const baseProps = {
    title: 'title',
    children: null,
  }

  afterEach(cleanup);

  it('renders correctly', () => {
    const rendered = render(<InfoSection {...baseProps} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
