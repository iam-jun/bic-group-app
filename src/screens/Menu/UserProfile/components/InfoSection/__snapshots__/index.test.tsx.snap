// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoSection component renders correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 16,
      "paddingBottom": 12,
    }
  }
  testID="info_section"
>
  <Text
    style={
      Array [
        Object {
          "color": undefined,
          "fontFamily": "BeVietnamPro-SemiBold",
          "fontSize": 17,
        },
        Object {},
        Object {
          "marginBottom": 16,
        },
      ]
    }
    testID="info_section.title"
  />
</View>
`;
